### ISSUE TITLE
Various Improvement suggestions

### ISSUE THREAD

**Issue by nileplumb** [versx/WhMgr-UI#39]
- Fix lures sections (seems wonky for me, buttons have no label etc) + limit by Area, Distance or Pokéstop name
- Gym Raid Subscription not doing eggs? perhaps rework the ui gym sub to be able to select egg tier, raid tier, or specific mon + limit by Area, Distance or Gym Name (perhaps show gym's image to help with duplicates)
- Choosing areas fences from a clickable map would be v user friendly!
- Setting a users location via a 'locate me' button (perhaps even 2+ locations? eg work/home, could expand to a schedule switch etc)
- Showing a mini map of a users saved location + default distance spread would be useful.
- Mention any quirks about subs. eg PvP is wild spawns, max level 50.
- Visual Pokémon form selection boxes + all forms selection button
- Invasions should list Grunt Types rather than Pokémon + limit by Area, Distance or Pokéstop name (show Pokéstop's image to help with duplicates)
- Quest subs should be visual rather than Search + Dropdown + limit by Area, Distance or Pokéstop name
- Admin customisable 'presets' buttons for Pokémon Subs eg XL Karp, XS Rat, PvP Cup Meta, Dragon Types, whatever becomes relevant in future.

**Comment by versx** [comment]
- Run `npm run update` or `npm run create-locales`
- No raids have never supported eggs subs, although it will eventually, especially with the new `gymme` subscription changes
- Absolutely would be nice, thanks for the suggestion
- I do have plans to add a locations rework to support multiple instead of just one, I like the locate me button though
- Absolutely would be nice, thanks for the suggestion
- Honestly, it's easier imo if people just make a PR for an FAQ to the read the docs page or add it to an existing page, I can't be bothered with it, markdown syntax is simple enough. Also, PvP stats are based on your backend support AFAIK
- Although doable, I'd kind of rather not, looking up each pokemons form name could be taxing, although I'll do some tests of course if I implement it
- Invasion subs will receive a rework too to add back grunt type subs (which it was originally based on, not rewards), although the rework will **not** support discord commands and only via WhMgr-UI
- Doing distances per sub isn't a problem when setting them via the UI, the issue is really via Discord commands which will turn very ugly for the layman user, which of course we could set a limited amount of features for Discords command subscriptions as well as really just removing them entirely or making it optional to remove them I guess.
- Nice suggestion, I'll try and remember to add something like this

Also, half of this should have been for WhMgr vs WhMgr-UI, since WhMgr-UI just manipulates the database so WhMgr can read the changes.

**Comment by versx** [comment]
- [x] Fix lures sections (seems wonky for me, buttons have no label etc) + limit by Area, Distance or Pokéstop name
- [x] Gym Raid Subscription not doing eggs? perhaps rework the ui gym sub to be able to select egg tier, raid tier, or specific mon + limit by Area, Distance or Gym Name (perhaps show gym's image to help with duplicates)
- [x] Choosing areas fences from a clickable map would be v user friendly!
- [x] Setting a users location via a 'locate me' button (perhaps even 2+ locations? eg work/home, could expand to a schedule switch etc)
- [x] Showing a mini map of a users saved location + default distance spread would be useful.
- [x] Mention any quirks about subs. eg PvP is wild spawns, max level 50.
- [ ] Visual Pokémon form selection boxes + all forms selection button
- [x] Invasions should list Grunt Types rather than Pokémon + limit by Area, Distance or Pokéstop name (show Pokéstop's image to help with duplicates)
- [ ] Quest subs should be visual rather than Search + Dropdown + limit by Area, Distance or Pokéstop name
- [ ] Admin customizable 'presets' buttons for Pokémon Subs eg XL Karp, XS Rat, PvP Cup Meta, Dragon Types, whatever becomes relevant in future.

**Comment by versx** [comment]
@nileplumb
- Regarding `Visual Pokemon form selection boxes` how do you suggest displaying forms, since of course they are Pokemon specific, just icons that say Alola, Galar, A, B , C, Party Hat, etc?
- Quest subs should be visual -> Just show every combination of possible quests or only what's in current database possibly?
- Admin customizable 'presets' buttons: just thinking how this would look in the config or another external file. 🤔 would love to implement though.

**Comment by nileplumb** [comment]
well I was thinking it would act like a filter of sorts, so it would check which pokemon you've selected then only display the icons (or text if icon missing) for the available forms of the selected pokemon.
because at the moment all possible forms are listed including some confusing duplicates

 eg 
upcoming Slowpoke people might erroneously select a 'normal' Slowpoke or 'Galar Standard', when in fact this would be normal castform and darmanitan

I'd argue against a drop down box perhaps a collapsible section, default All forms selected (including future releases not yet in master file eg a pikachu subscription should alert for any weird form costumes they release in future )


QUESTS
I believe it should show every possible combination, at the moment many people are chasing Galarian Zigzagoon and Ponyta quests, let's assume a player wanted to set up their alerts now for a future event they can't, nor can they if they try to set up when the database is quest empty (eg with so many events now I often delay quest scans until 10am on new event days, the database would be empty and players can't chose any new subscriptions)

ADMIN PRESETS SUBSCRIPTION BUTTONS
are separate json files not an option?

eg
{
	"min_iv": 0,
	"max_iv": 100,
	"min_level": 0,
	"max_level": 35,
	"gender": "all",
	"size": "all",

	"Exeggutor": {
		"form": Alola, (or specific Id number eg 78)
		"min_iv": 0,
		"max_iv": 100,
		"min_level": 0,
		"max_level": 35,
		"size": "All",
		"gender": "All"
	},
	"Dratini": "True",
	"Dragonair": "True",
	"Dragonite": "True",
	"Kingdra": "True",
	"Altaria": "True",
	"Bagon": "True",
	"Shelgon": "True",
	"Salamence": "True",
	"Latias": "True",
	"Latios": "True",
	"Rayquaza": "True",
	"Gible": "True",
	"Gabite": "True",
	"Garchomp": "True",
	"Dialga": "True",
	"Palkia": "True",
	"Giratina": "True"
}

**Comment by nileplumb** [comment]
another good option that'd be very user friendly would be for a logged in admin to chose the pokemon themselves then have an option to save as a new preset for others choosing a button name etc

**Comment by versx** [comment]
> well I was thinking it would act like a filter of sorts, so it would check which pokemon you've selected then only display the icons (or text if icon missing) for the available forms of the selected pokemon.
> because at the moment all possible forms are listed including some confusing duplicates
> 
> eg
> upcoming Slowpoke people might erroneously select a 'normal' Slowpoke or 'Galar Standard', when in fact this would be normal castform and darmanitan
> 
> I'd argue against a drop down box perhaps a collapsible section, default All forms selected (including future releases not yet in master file eg a pikachu subscription should alert for any weird form costumes they release in future )
> 
> QUESTS
> I believe it should show every possible combination, at the moment many people are chasing Galarian Zigzagoon and Ponyta quests, let's assume a player wanted to set up their alerts now for a future event they can't, nor can they if they try to set up when the database is quest empty (eg with so many events now I often delay quest scans until 10am on new event days, the database would be empty and players can't chose any new subscriptions)
> 
> ADMIN PRESETS SUBSCRIPTION BUTTONS
> are separate json files not an option?
> 
> eg
> {
> "min_iv": 0,
> "max_iv": 100,
> "min_level": 0,
> "max_level": 35,
> "gender": "all",
> "size": "all",
> 
> ```
> "Exeggutor": {
> 	"form": Alola, (or specific Id number eg 78)
> 	"min_iv": 0,
> 	"max_iv": 100,
> 	"min_level": 0,
> 	"max_level": 35,
> 	"size": "All",
> 	"gender": "All"
> },
> "Dratini": "True",
> "Dragonair": "True",
> "Dragonite": "True",
> "Kingdra": "True",
> "Altaria": "True",
> "Bagon": "True",
> "Shelgon": "True",
> "Salamence": "True",
> "Latias": "True",
> "Latios": "True",
> "Rayquaza": "True",
> "Gible": "True",
> "Gabite": "True",
> "Garchomp": "True",
> "Dialga": "True",
> "Palkia": "True",
> "Giratina": "True"
> ```
> 
> }

**Forms:** I like the idea of only showing the available form buttons/selections relating to the Pokemon selected.

**Quests:** Thanks

**Admin Presets:** Ok so using PokeAlarm filtering for the standard for presets, can probably do that.

**Comment by nileplumb** [comment]
yeah customisable presets would be more of an advanced and optional thing deployed by a server admin, (though including XL karp and Tiny Rat presets would be welcome by many ) I think some filter configuration by json is acceptable for them.
